tensorflow/core/kernels/random_op.cc

```diff
@@ -613,50 +598,35 @@ struct FillPhiloxRandomKernel<Distribution, false> {
 template <class Distribution>
 struct FillPhiloxRandomKernel<Distribution, true> {
   typedef typename Distribution::ResultElementType T;
-  using write_accessor = sycl::accessor<uint8_t, 1, sycl::access::mode::write,
-                                        sycl::access::target::global_buffer>;
+  using write_accessor =
+    cl::sycl::accessor<uint8_t, 1, cl::sycl::access::mode::write,
+                       cl::sycl::access::target::global_buffer>;
 
   FillPhiloxRandomKernel(write_accessor& data, random::PhiloxRandom& gen,
                          Distribution& dist)
       : data_(data), gen_(gen), dist_(dist) {}
 
-  void operator()(sycl::nd_item<1> item) {
+  void operator()(cl::sycl::nd_item<1> item) {
     using random::PhiloxRandom;
     using random::SingleSampleAdapter;
 
-    const size_t kReservedSamplesPerOutput = 256;
-    const size_t kGroupSize = Distribution::kResultElementCount;
-    const size_t kGeneratorSkipPerOutputGroup =
-        kGroupSize * kReservedSamplesPerOutput /
-        PhiloxRandom::kResultElementCount;
+    constexpr size_t kReservedSamplesPerOutput = 256;
```

Comment:
Not that I know of. It's the constant used for CPU and for CUDA. And yeah because of the way the kernel are built we have to duplicate it for each different device.

Reply:
So shouldn't we have a global static variable to store this value if it's going to be used at other places?